### ISSUE TITLE
Fix Request: AvoidInfiniteRecursion false positive for same number of params, other types

### ISSUE THREAD

**Issue by stokpop** [jborgers/PMD-jPinpoint-rules#498]
Example:

```java
    public void tellNext(TbMsg msg, String relationType) {
        tellNext(msg, Collections.singleton(relationType)); // <-- false positive
    }

    public void tellNext(TbMsg msg, Set<String> relationTypes) {
       ... 
        nodeCtx.tell(new TellNextMsg(msg, relationTypes));
    }
```

Possible solution: when multiple methods with same number of params in same class, assume it is no recursion

More cases with singe/collection, e.g. feature/feature**s**

```java
    public Map<Feature, Value> checkStateUpdated(Set<Feature> features) {
        Map<Feature, Value> result = new HashMap<>();
        for (Feature feature : features) {
            Pair<Feature, Value> tmp = checkStateUpdated(feature);
        }
        return result;
    }

    private Pair<Feature, Value> checkStateUpdated(Feature feature) {
        Value value = createValue(feature);
        return Pair.of(feature, value);
    }

```
